### PR TITLE
Separate library from other ArtifactTypes

### LINKED ISSUE #202: Refactor REST / Shell API to treat ArtifactType.library on its own
As a developer, I'd like to refactor `rest` and `shell` APIs to adapt to `ArtifactType.library` support.

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes #202

### CODE REVIEW

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/LibraryTemplate.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+	/**
+	 * Template for URI creation.
+	 */
+	private final UriTemplate uriTemplate;
+
+	/**
+	 * Construct a {@code LibraryTemplate} object.
+	 * @param restTemplate    template for HTTP/rest commands
+	 * @param resourceSupport HATEOS link support
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/LibraryTemplate.java]: HATEOAS ;)

File: spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/LibraryCommands.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	@CliAvailabilityIndicator({REGISTER_LIBRARY, UNREGISTER_LIBRARY, LIST_LIBRARIES})
+	public boolean available() {
+		return dataFlowShell.getDataFlowOperations() != null;
+	}
+
+	@CliCommand(value = REGISTER_LIBRARY, help = "Register a new library")
+	public String register(
+			@CliOption(mandatory = true,
+					key = {"", "name"},
+					help = "the name for the registered library")
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/LibraryCommands.java]: may be `the name for the library to register`

**Comment by ericbottard** [comment]
Addressed review comments

File: spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/LibraryCommands.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+			@CliOption(mandatory = true,
+					key = {"coordinates"},
+					help = "coordinates to the library")
+			String coordinates,
+			@CliOption(key = "force",
+					help = "force update if library already exists (only if not in use)",
+					specifiedDefaultValue = "true",
+					unspecifiedDefaultValue = "false")
+			boolean force) {
+		libraryOperations().register(name, coordinates, force);
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/LibraryCommands.java]: If the `registration` operation returns when the library already exists and the `force` is `false`, then we need to display appropriate message as the `Successful` message might be misleading. Will create a separate issue for this.

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Comment by ilayaperumalg** [comment]
Squashed, rebased and merged as 35be96e6e45915ec5b4986ab9630578d4054e0aa

**Comment by ilayaperumalg** [comment]
Created https://github.com/spring-cloud/spring-cloud-dataflow/issues/266 to track [this] (https://github.com/spring-cloud/spring-cloud-dataflow/pull/229#discussion_r48077209)

**Pull request closed without merging**